vendors/nuvoton/boards/numaker_iot_m487_wifi/ports/pkcs11/core_pkcs11_pal.c

```diff
@@ -563,29 +676,42 @@ CK_RV PKCS11_PAL_DestroyObject( CK_OBJECT_HANDLE xHandle )
 
     if( xResult == CKR_OK )
     {
-        /* Some ports return a pointer to memory for which using memset directly won't work. */
-        pxZeroedData = pvPortMalloc( ulObjectLength );
-
-        if( NULL != pxZeroedData )
+        if( ulObjectLength > 0 )
         {
-            /* Zero out the object. */
-            ( void ) memset( pxZeroedData, 0x0, ulObjectLength );
+            /* Some ports return a pointer to memory for which using memset directly won't work. */
+            pxZeroedData = pvPortMalloc( ulObjectLength );
```

Comment:
To destroy object by prvFLASH_DeleteFile, it's not necessary to prepare pxZeroedData to zero out the object.

```suggestion

```